include/openthread/border_agent.h

```diff
@@ -180,117 +161,138 @@ otError otBorderAgentGetId(otInstance *aInstance, otBorderAgentId *aId);
+ * Represents Border Agent's Ephemeral Key Manager state.
+ */
+typedef enum otBorderAgentEphemeralKeyState
+{
+    OT_BORDER_AGENT_STATE_DISABLED  = 0, ///< Ephemeral Key Manager is disabled.
+    OT_BORDER_AGENT_STATE_STOPPED   = 1, ///< Enabled, but no ephemeral key is in use (not set or started).
+    OT_BORDER_AGENT_STATE_STARTED   = 2, ///< Ephemeral key is set. Listening to accept secure connections.
+    OT_BORDER_AGENT_STATE_CONNECTED = 3, ///< Session is established with an external commissioner candidate.
+    OT_BORDER_AGENT_STATE_ACCEPTED  = 4, ///< Session is established and candidate is accepted as full commissioner.
+} otBorderAgentEphemeralKeyState;
+
+/**
+ * Gets the state of Border Agent's Ephemeral Key Manager.
  *
- * The Ephemeral Key feature can only be used when it's enabled. This information will be displayed in a bitmap in the
- * txt records of the meshcop service published by this Border Router.
+ * Requires `OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE`.
  *
- * The default value is `OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_FEATURE_ENABLED_BY_DEFAULT`.
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
  *
- * @param[in] aInstance    The OpenThread instance.
+ * @returns The current state of Ephemeral Key Manager.
  */
-bool otBorderAgentIsEphemeralKeyFeatureEnabled(otInstance *aInstance);
+otBorderAgentEphemeralKeyState otBorderAgentEphemeralKeyGetState(otInstance *aInstance);
 
 /**
- * Enables/disables the Border Agent Ephemeral Key feature.
+ * Enables/disables the Border Agent's Ephemeral Key Manager.
+ *
+ * Requires `OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE`.
  *
- * If an ephemeral key is already active and then this method is called to disable the feature, the in-use ephemeral
- * key will be cleared.
+ * If this function is called to disable, while an an ephemeral key is in use, the ephemeral key use will be stopped
+ * (as if `otBorderAgentEphemeralKeyStop()` is called).
  *
  * @param[in] aInstance    The OpenThread instance.
- * @param[in] aEnabled     Whether to enable the BA Ephemeral Key feature.
+ * @param[in] aEnabled     Whether to enable or disable the Ephemeral Key Manager.
  */
-void otBorderAgentSetEphemeralKeyFeatureEnabled(otInstance *aInstance, bool aEnabled);
+void otBorderAgentEphemeralKeySetEnabled(otInstance *aInstance, bool aEnabled);
 
 /**
- * Sets the ephemeral key for a given timeout duration.
+ * Starts using an ephemeral key for a given timeout duration.
  *
  * Requires `OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE`.
  *
- * The API SHOULD only be called when the Ephemeral Key feature is enabled (which can be set by
- * `otBorderAgentSetEphemeralKeyFeatureEnabled`) or configured as enabled by default.
+ * An ephemeral key can only be set when `otBorderAgentEphemeralKeyGetState()` is `OT_BORDER_AGENT_STATE_STOPPED`,
+ * i.e., enabled but not yet started; otherwise, `OT_ERROR_INVALID_STATE` is returned. This means that setting the
```

Comment:
```suggestion
 * i.e., enabled but not yet started. Otherwise, `OT_ERROR_INVALID_STATE` is returned. This means that setting the
```

Avoid semi colons if possible